cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java

```diff
@@ -99,27 +93,19 @@ public void extractTemplateSetsTest() throws IOException {
   @Test
   public void extractTemplatesWithOldConfiguration() throws IOException {
 
-    Path cobigenTemplatesParent = this.cobiGenHome.resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATES_PATH);
-    Files.createDirectories(cobigenTemplatesParent);
+    FileUtils.copyDirectory(new File(testFileRootPathMonolithicTemplates),
+        this.cobiGenHomeMonolithicTemplates.toFile());
+
+    Path cobigenTemplatesParent = this.cobiGenHomeMonolithicTemplates
+        .resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATES_PATH);
+
     Path cobigenTemplatesProject = cobigenTemplatesParent.resolve(ConfigurationConstants.COBIGEN_TEMPLATES);
-    Files.createDirectories(cobigenTemplatesProject);
-    CobiGenFactory.extractTemplates();
-    assertThat(cobigenTemplatesProject).exists().isDirectory();
-  }
 
-  /**
-   * Test of find template set downloaded folder to ensure backwards compatibility
-   *
-   * @throws IOException if an Exception occurs
-   */
-  @Test
-  public void findTemplateSetsDownloadedFolderTest() throws IOException {
+    TemplateAdapter templateAdapter = new TemplateAdapterImpl(cobigenTemplatesParent);
+    templateAdapter.adaptMonolithicTemplates(cobigenTemplatesProject, false);
 
-    Path downloadedFolder = this.cobiGenHome.resolve("template-sets").resolve("downloaded");
-    Files.createDirectories(downloadedFolder);
-    URI templatesLocationURI = ConfigurationFinder.findTemplatesLocation();
-    Path expectedDownloadedFolder = Paths.get(templatesLocationURI).resolve("downloaded");
-    assertThat(expectedDownloadedFolder).exists().isDirectory();
+    assertThat(cobigenTemplatesProject).exists().isDirectory();
+    assertThat(cobigenTemplatesProject.resolve("src/main/templates")).exists().isDirectory();
```

Comment:
Please use: ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER instead of the fixed string.